src/components/UI/Pagination/Pagination.tsx

```diff
@@ -21,32 +25,71 @@ export const Pagination: FunctionComponent<PaginationProps> = ({ totalNoOfPages,
     }
   };
 
+  const goToFirstPage = (): void => {
+    setCurrentPage(1);
+  };
+
+  const goToLastPage = (): void => {
+    setCurrentPage(totalNoOfPages);
+  };
+
+  const populateNumbers = (index: number): number | string => {
+    switch (true) {
+      case (currentPage < 4 && index < 5) || index + 1 === totalNoOfPages:
+      case currentPage >= 4 && currentPage < totalNoOfPages - 4 && index >= currentPage - 3 && index <= currentPage + 1:
+      case currentPage >= totalNoOfPages - 4 && index >= totalNoOfPages - 7:
+        return index + 1;
+
+      default:
+        return "...";
+    }
+  };
+
+  const generatePageNumbers = (): ReactElement[] => {
+    const pagesArray = [...Array(totalNoOfPages)].map((pages, i) => (!expandPagination ? populateNumbers(i) : i + 1));
+
```

Comment:
can expandPagination works in such a way that expands only the left portion when click on the left truncation? same for right side

Reply:
ok